File: src/python/core/src/dicebear/style_def/meta_source.py
"""Read-only view over the ``meta.source`` block of a style definition."""

from __future__ import annotations

from typing import Any, cast


class MetaSource:
    """Read-only view over the ``meta.source`` block of a style definition."""

    def __init__(self, data: dict[str, Any]) -> None:
        self._data = data

    def name(self) -> str | None:
        """Return the source name (e.g. the original work title), or ``None``."""
        return cast("str | None", self._data.get("name"))

    def url(self) -> str | None:
        """Return the URL of the source, or ``None`` when not set."""
        return cast("str | None", self._data.get("url"))

File: src/csharp/core/src/Style/MetaSource.cs
using System.Text.Json.Nodes;

namespace DiceBear.Internal
{
    /// <summary>
    /// Read-only view over the <c>meta.source</c> block of a style definition.
    /// </summary>
    internal sealed class MetaSource
    {
        private readonly JsonObject _data;

        internal MetaSource(JsonObject data) => _data = data;

        /// <summary>
        /// Returns the source name (e.g. the original work title), or
        /// <see langword="null"/> when not set.
        /// </summary>
        internal string? Name() => JsonRead.Str(_data, "name");

        /// <summary>
        /// Returns the URL of the source, or <see langword="null"/> when not
        /// set.
        /// </summary>
        internal string? Url() => JsonRead.Str(_data, "url");
    }
}
